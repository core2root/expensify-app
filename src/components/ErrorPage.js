import React from 'react';
import {Link} from 'react-router-dom';

const ErrorPage =()=>(
  <div>
    <p>Oops! looks like this page doesen't exist <b>CODE: 404</b> </p>
    <Link to='/'>Go Home</Link>
  </div>
);

export default ErrorPage;