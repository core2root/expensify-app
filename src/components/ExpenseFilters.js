import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, setSortByDate, setSortByAmount, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import 'react-dates/initialize';

class ExpenseFilters extends React.Component{

  state = {
    calenderFocused: null,
    startDateId:'123_fg_f3',
    endDateId:'bb_097_bdsd'
  };

  onDatesChange = ({startDate, endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  }

  onFocusChange = (calenderFocused) => {
    this.setState(() => ({calenderFocused}))
  };

  render() {
    return (
      <div>
        <input
          type='text'
          value={this.props.filters.text} onChange={(e) => {
            this.props.dispatch(setTextFilter(e.target.value))
          }
          } />
        <select
          value={this.props.filters.sortBy == 'date' ? 'date' : 'amount'}
          onChange={(e) => {
            const val = e.target.value;
            this.props.dispatch(val == 'date' ? setSortByDate() : setSortByAmount());
          }
          }>
          <option value='date'>By Date</option>
          <option value='amount'>By Amount</option>
        </select>
        <DateRangePicker 
          startDate = {this.props.filters.startDate}
          endDate = {this.props.filters.endDate}
          onDatesChange = {this.onDatesChange} 
          focusedInput = {this.state.calenderFocused} 
          onFocusChange = {this.onFocusChange}
          endDateId={this.state.endDateId}
          startDateId = {this.state.startDateId}
          numberOfMonths = {1}
          isOutsideRange = {() => false}
          showClearDates = {true}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
}

export default connect(mapStateToProps)(ExpenseFilters);