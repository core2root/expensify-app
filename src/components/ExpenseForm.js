import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/initialize'; 

export default class ExpenseForm extends React.Component{
  constructor(props){
    super(props);

    const expense = props.expense;
    this.state = {
      description: expense ? expense.description : '',
      note: expense ? expense.note : '',
      amount: expense ? (expense.amount/100).toString() : '',
      createdAt: expense ? moment(expense.createdAt) : moment(),
      calenderFocused: false,
      error: undefined
    };
  }

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({description}));
  };

  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({note}));
  };

  onAmountChange = (e) => {
    const amount = e.target.value;
    if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
      this.setState(() => ({amount}));
    }
  };

  onDateChange = (createdAt) => {
    if(createdAt)
      this.setState(() => ({createdAt}));
  }

  onFocusChange = ({focused}) => {
    this.setState(() => ({calenderFocused: focused}));
  }

  onSubmit = (e) => {
    e.preventDefault();
    if(!this.state.description || !this.state.amount){
      //Set error state = Please provide description and amount
      this.setState(() => ({error: 'Please provide description and amount'}));
    }else{
      //Clear error
      this.setState(() => ({error: undefined}));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  }

  render(){
    return(
      <div>
        <form onSubmit = {this.onSubmit}>
          {this.state.error && <p>{this.state.error}</p>}
          <input 
            type='text' 
            placeholder='Description' 
            autoFocus 
            value={this.state.description}
            onChange={this.onDescriptionChange}/>

          <input 
            type='number' 
            placeholder='amount'
            value={this.state.amount}
            onChange={this.onAmountChange}/>

          <SingleDatePicker 
            date={this.state.createdAt}
            numberOfMonths = {1}
            onDateChange={this.onDateChange}
            focused = {this.state.calenderFocused}
            onFocusChange = {this.onFocusChange}
            isOutsideRange={(day) => false }
          />

          <textarea
            placeholder='Add a note for this expense (optional)'
            value={this.state.note}
            onChange={this.onNoteChange}>
          </textarea>

          <button>Add Expense</button>

        </form>
      </div>
    );
  }
}