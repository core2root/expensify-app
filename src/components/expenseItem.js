import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseItem = ({ id, description, amount, createdAt}) => (
  <div>
    <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
    <p>
      Amount: {numeral(amount/100).format('$0,0.00')} -
      Created At: {moment(createdAt).format('Do MMMM, YYYY')}
    </p>
  </div>
);

export default ExpenseItem; 
