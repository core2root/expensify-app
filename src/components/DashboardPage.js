import React from 'react';
import ExpenseList from './expenseList';
import ExpenseFilters from './ExpenseFilters';
import ExpensesSummary from './ExpensesSummary';

const DashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseFilters />
    <ExpenseList />
  </div>
);

export default DashboardPage;