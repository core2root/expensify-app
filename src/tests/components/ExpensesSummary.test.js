import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';


test('should render correctly expenses summary with 1 expense', () => {
 const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={25.36} />);
 expect(wrapper).toMatchSnapshot();
});

test('should render correctly expenses summary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={25} expensesTotal={2565.76} />);
  expect(wrapper).toMatchSnapshot();
});