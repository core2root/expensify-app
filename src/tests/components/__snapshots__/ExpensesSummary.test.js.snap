// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly expenses summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     total 
    $0.25
  </h1>
</div>
`;

exports[`should render correctly expenses summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    25
     
    expenses
     total 
    $25.66
  </h1>
</div>
`;
